Controllers/users.js

```diff
@@ -1,10 +1,34 @@
+const bcrypt = require("bcryptjs");
+const { isEmail, isStrongPassword } = require("validator");
+
 const User = require("../Models/users");
 
 const signin = async (req, res) => {
   res.json({ message: "Signed in!" });
 };
+
 const signup = async (req, res) => {
-  res.json({ message: "Signed up!" });
+  const { email, password } = req.body;
+
+  try {
+    if (!email) throw Error("Email is required!");
+    if (!isEmail(email)) throw new Error("Email is not valid.");
+
+    if (!password) throw Error("Password is required!");
+    if (!isStrongPassword(password)) throw new Error("Password is not strong.");
```

Comment:
Do you have the option to set the conditions for what is a `strongPassword`? I cannot see, what the user should provide to pass this test.

Reply:
Right, validator has a default set of options, but that is a little bit of an overkill if you ask me:

```javascript
{ 
minLength: 8, minLowercase: 1, minUppercase: 1, minNumbers: 1,
minSymbols: 1, returnScore: false, pointsPerUnique: 1, pointsPerRepeat: 0.5,
pointsForContainingLower: 10, pointsForContainingUpper: 10,
pointsForContainingNumber: 10, pointsForContainingSymbol: 10 }
```

Should I return a better error message containing the information about requirements or lower the standarts to `minLength: 8`?